tests/Behat/Context/Setup/ProductContext.php

```diff
@@ -0,0 +1,132 @@
+        SlugGeneratorInterface $slugGenerator,
+        JavaScriptTestHelperInterface $testHelper,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->productFactory = $productFactory;
+        $this->productRepository = $productRepository;
+        $this->productVariantFactory = $productVariantFactory;
+        $this->productVariantRepository = $productVariantRepository;
+        $this->productTranslationFactory = $productTranslationFactory;
+        $this->productTranslationRepository = $productTranslationRepository;
+        $this->localeContext = $localeContext;
+        $this->slugGenerator = $slugGenerator;
+        $this->testHelper = $testHelper;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @Given there is a product item with identifier :arg1
+     */
+    public function thereIsAProductItemWithIdentifier($code)
+    {
+        /** @var ProductInterface $productItem */
+        $productItem = $this->productFactory->createNew();
+        $productItem->setCode($code);
+        $productItem->setCreatedAt(new \DateTime());
+        $productItem->setEnabled(true);
+        $productItem->setVariantSelectionMethod('match');
+        $this->productRepository->add($productItem);
+
+        /** @var ProductVariantInterface $productVariant */
+        $productVariant = $this->productVariantFactory->createNew();
+        $productVariant->setName('test');
+        $productVariant->setCode($code);
+        $productVariant->setPosition(1);
+        $productVariant->setProduct($productItem);
+        $productVariant->setEnabled(true);
+        $this->productVariantRepository->add($productVariant);
+
+        /** @var ProductTranslationInterface $productTranslation */
+        $productTranslation = $this->productTranslationFactory->createNew(ProductTranslationInterface::class);
+        $productTranslation->setName($code);
+        $productTranslation->setSlug($this->slugGenerator->generate($code));
+        $productTranslation->setTranslatable($productItem);
+        $productTranslation->setLocale($this->localeContext->getLocaleCode());
+        $this->productTranslationRepository->add($productTranslation);
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully enqueued
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyEnqueued()
+    {
+        $this->testHelper->waitUntilNotificationPopups(
+            $this->notificationChecker, NotificationType::success(), 'Akeneo PIM product import has been successfully scheduled'
+        );
+    }
+
+    /**
+     * @Given /^I should be notified that it has been already enqueued$/
+     */
+    public function iShouldBeNotifiedThatItHasBeenAlreadyEnqueued()
+    {
+        $this->testHelper->waitUntilNotificationPopups(
+            $this->notificationChecker, NotificationType::success(), 'Akeneo PIM import for this product has been already scheduled before'
+        );
+    }
```

Comment:
These context steps should be in a "Ui" context and not in a "Setup" context. I'd move in the `ManagingProducts` context.